File: scheduler/core/src/testFixtures/java/io/orkes/conductor/scheduler/service/MockExecutor.java
package io.orkes.conductor.scheduler.service;

public class MockExecutor implements SchedulerServiceExecutor {

    private MockExecutorService executorServiceMainQueuePoll = new MockExecutorService();
    private MockExecutorService executorServiceArchivalQueuePoll = new MockExecutorService();

    @Override
    public MockExecutorService getExecutorServiceMainQueuePoll(int poolSize) {
        return executorServiceMainQueuePoll;
    }

    @Override
    public MockExecutorService getExecutorServiceArchivalQueuePoll(int poolSize) {
        return executorServiceArchivalQueuePoll;
    }
}

File: scheduler/core/src/testFixtures/typescript/io/orkes/conductor/scheduler/service/MockExecutor.ts
import { MockExecutorService } from './MockExecutorService';

export class MockExecutor {
    private executorServiceMainQueuePoll: MockExecutorService = new MockExecutorService();
    private executorServiceArchivalQueuePoll: MockExecutorService = new MockExecutorService();

    public getExecutorServiceMainQueuePoll(poolSize: number): MockExecutorService {
        return this.executorServiceMainQueuePoll;
    }

    public getExecutorServiceArchivalQueuePoll(poolSize: number): MockExecutorService {
        return this.executorServiceArchivalQueuePoll;
    }
}
